### PR TITLE
script_score query errors on negative scores

### PR DESCRIPTION
**Pull request by mayya-sharipova**
7.5 and 7.6 had a regression that allowed for
script_score queries to have negative scores.
We have corrected this regression in #52478.
This is an addition to #52478 that adds
a test and release notes.

### CODE REVIEW

**Review by jtibshirani — COMMENTED** [review]
@mayya-sharipova thanks for making this update!

For the test, I think it'd be nice to add a case to `ScriptScoreQueryTests` instead of adding a new `yml` REST test. The behavior seems pretty straightforward to cover through a unit test, and unit tests tend to be faster to run and easier to debug than integration tests.

**Comment by mayya-sharipova** [comment]
@jtibshirani Thanks for the review, I have addressed your comments in the second commit. Please continue to review. 
I thought `yml` test is more understandable and it highlights versions we made changes, while in unit tests with scripts we do a lot of mocks.

File: server/src/test/java/org/elasticsearch/search/query/ScriptScoreQueryTests.java [hunk 1]
```diff
@@ -131,6 +131,14 @@ public void testExplainDefaultNoScore() throws IOException {
         assertThat(explanation.getValue(), equalTo(2.0f));
     }
 
+    public void testScriptScoreErrorOnNegativeScore() {
```
Comment (jtibshirani) [inline server/src/test/java/org/elasticsearch/search/query/ScriptScoreQueryTests.java]: For me this set of unit tests is quite straightforward + easy to understand. I think yml tests are slower to debug and are most helpful for testing the end-to-end integration. We recently added a set of testing guidelines that I think are really helpful: https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc#what-kind-of-tests-should-i-write

**Review by jtibshirani — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/search/query/ScriptScoreQueryTests.java (hunk 1 above)
Comment (mayya-sharipova) [reply server/src/test/java/org/elasticsearch/search/query/ScriptScoreQueryTests.java]: @jtibshirani Thanks a lot, I missed these guidelines, will study them

**Pull request merged**